jsapp/js/components/modals/koboModal.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import Modal from 'react-modal';
+import bem, {makeBem} from 'js/bem';
+import './koboModal.scss';
+
+bem.KoboModal = makeBem(null, 'kobo-modal');
+
+type KoboModalSize = 'large' | 'medium';
+
+interface KoboModalProps {
+  /** For displaying the modal. */
+  isOpen: boolean;
+  /** Request from the inside for the modal parent to close it. */
+  onRequestClose: () => void;
+  size?: KoboModalSize;
+  children: React.ReactNode;
+  /**
+   * Whether it should close when user hits Esc or clicks on overlay.
+   * NOTE: disabling Esc key may introduce an accessibility issue.
+   */
+  isDismissableByDefaultMeans?: boolean;
+  /** Renders a `data-testid` attribute in the DOM. */
+  testId?: string;
```

Comment:
Was there any conversation about this? It seems like a new concept. Why use testId over id?

Reply:
Is it `data-cy` with a new name?

Reply:
There is a built-in `testId` property in `react-modal` and I thought it would be better to use it instead of not having any cypress selector. But I did find a way to use `data-cy` with it :)